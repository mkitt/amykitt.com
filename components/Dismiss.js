// @flow
import React from 'react'
import css, { hover, media3 } from '../styles/css'
import Button from './Button'
import { XIcon } from './Icons'

type Props = {
  className?: string,
  title?: string,
}

const style = css(
  {
    position: 'absolute',
    top: 24,
    right: 24,
    zIndex: 2,
    color: '#7c7c7c',
    transition: 'color 0.2s',
  },
  hover({ color: '#000' }),
  media3({ position: 'fixed' }),
)

const Dismiss = ({ className, title, ...props }: Props) => (
  <Button
    className={`${className || ''} ${style}`}
    title={title || 'Dismiss'}
    {...props}
  >
    <XIcon />
  </Button>
)

Dismiss.defaultProps = {
  className: '',
  title: null,
}

export default Dismiss
