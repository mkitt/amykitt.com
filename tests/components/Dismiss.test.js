import React from 'react'
import renderer from 'react-test-renderer'
import Dismiss from '../../components/Dismiss'

const noop = () => {}

describe('Dismiss', () => {
  let component

  beforeEach(() => {
    component = (
      <Dismiss
        className="DismissClassName"
        onClick={noop}
      />
    )
  })

  it('shows the correct snapshot for the Dismiss component tree', () => {
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
