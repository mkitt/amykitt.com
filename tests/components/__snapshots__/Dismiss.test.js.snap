// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dismiss shows the correct snapshot for the Dismiss component tree 1`] = `
<button
  className="css-1fqszn3 DismissClassName css-rh1ike"
  onClick={[Function]}
  title="Dismiss"
  type="button"
>
  <svg
    aria-hidden={true}
    height={24}
    viewBox="0 0 24 24"
    width={24}
  >
    <g
      stroke="currentColor"
    >
      <line
        x1="2"
        x2="22"
        y1="2"
        y2="22"
      />
      <line
        x1="22"
        x2="2"
        y1="2"
        y2="22"
      />
    </g>
  </svg>
</button>
`;
